pandas/_config/__init__.py

```diff
@@ -31,10 +31,70 @@
 
 
 def using_copy_on_write() -> bool:
+    """
+    Determine if copy-on-write mode is being used.
+
+    This function examines the global pandas configuration settings.
+    Copy-on-write is a memory-saving technique that avoids copying data
+    until it is actually modified, which can be particularly useful when
+    working with large datasets.
+
+    Parameters
+    ----------
+        None
+
+    Returns
+    -------
+    bool
+        'True' if copy-on-write mode is enabled and the data manager
+        is set to "block", otherwise 'False'.
+
+    Example
+    -------
+    >>> pd.set_option('mode.chained_assignment', 'raise')
+    >>> pd.set_option('mode.copy', True)
+    >>> pd.set_option('mode.use_inf_as_na', True)
+    >>> pd.set_option('compute.use_bottleneck', False)
+    >>> pd.set_option('compute.use_numexpr', False)
+    >>> using_copy_on_write()
+    True
```

Comment:
there's a couple of issues here:
1. most of these options aren't relevant to this example
2. the options will stay set after the doctest for this function has finished (there have been issues due to this in the past)

How about we just put something like
```
>>> pd.set_option('mode.copy', True)
>>> using_copy_on_write()
True
>>> pd.unset_option('mode.copy')
>>> using_copy_on_write()
False
```
(I haven't tried running this, so the syntax might not be correct)